crypto/rand/drbg_lib.c

```diff
@@ -0,0 +1,436 @@
+/*
+ * Copyright 2011-2017 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the OpenSSL license (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <string.h>
+#include <openssl/crypto.h>
+#include <openssl/err.h>
+#include <openssl/rand.h>
+#include "rand_lcl.h"
+#include "rand_drbg_lcl.h"
+
+/*
+ * Support framework for SP800-90 DRBGs
+ */
+
+
+int drbg_fillin(DRBG_CTX *dctx, int type, unsigned int flags)
+{
+    int ret;
+
+    dctx->status = DRBG_STATUS_UNINITIALISED;
+    dctx->xflags = flags;
+    dctx->type = type;
+    dctx->iflags = 0;
+    dctx->entropy_blocklen = 0;
+    dctx->health_check_cnt = 0;
+    dctx->health_check_interval = DRBG_HEALTH_INTERVAL;
+
+    switch (type) {
+    default:
+        ret = -2;
+        break;
+    case NID_sha1:
```

Comment:
Is it worth considering not including the SHA1 based DRBGs?  This and hmacWithSHA1 below.

The known (and foreseeable) weakness in SHA1 is unlikely to impact its use as a DRBG but paranoia is bliss.